src/components/Shared/ScatterPlot/ScatterPlotComponent.tsx

```diff
@@ -21,17 +24,28 @@ export class ScatterPlotComponent extends LinePlotComponent {
         }
     }
 
-    private getScatterColor(value: number, min: number, max: number, toColor: number): string {
+    private getScatterColor(value: number, min: number, max: number, toColor: number, outRange: boolean): string {
         let percentage = (value + Math.abs(min)) / (Math.abs(min) + Math.abs(max));
         let hue = (percentage * toColor).toString(10);
-        return ["hsl(", hue, ",100%,50%)"].join("");
+        if (outRange) {
+            return ["hsla(", hue, ",0%,50%,", this.opacityOutRange, ")"].join("");
+        }
+        return ["hsla(", hue, ",100%,50%,", this.opacityInit, ")"].join("");
     }
 
     private fillColor(): Array<string> {
         let scatterColors = [];
-        if (this.props.data) {
-            this.props.data.forEach(data => {
-                let pointColor = this.getScatterColor(data.y, this.props.yMin, this.props.yMax, this.props.colorRangeEnd);
+        if (this.props.data && this.props.data.length && this.props.scatterColorIndex && this.props.scatterColorIndex.length && this.props.interactionBorder) {
+            let yMin = getMinY(this.props.data.filter(data => {return (!isNaN(data.x) && !isNaN(data.y)); }));
```

Comment:
this is a really inefficient way to get the min and max. I'm modifying the `getMinY`  and `getMaxY` functions to just use a normal for loop, and to ignore `NaN` values. It's a little less "neat", but much faster. See [here](https://jsperf.com/minmax-testing/1)